### PR TITLE
Inm/fix not updating in real time config values 2

### PR DESCRIPTION
**Pull request by irishwithaxe**
output files are writing according to run time changes of *interval config values

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/2151)

### CODE REVIEW

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by irishwithaxe** [comment]
Test!

**Comment by JustinPihony** [comment]
test!

**Comment by JustinPihony** [comment]
test!

**Comment by JustinPihony** [comment]
test!

File: src/main/scala/beam/router/BeamSkimmer.scala [hunk 1]
```diff
@@ -40,12 +39,15 @@ import scala.util.control.NonFatal
...
 //TODO to be validated against google api
 class BeamSkimmer @Inject()(
+  val beamServices: BeamServices,
   val beamScenario: BeamScenario,
   val geo: GeoUtils
 ) extends LazyLogging {
   import BeamSkimmer._
   import beamScenario._
 
+  def beamConfig: BeamConfig = beamServices.beamConfig
```
Comment (JustinPihony) [inline src/main/scala/beam/router/BeamSkimmer.scala]: There are still a number of places where `beamScenario.beamConfig` is used - I assume that should use the new `def`

File: src/main/scala/beam/sim/CustomTerminateAtFixedIterationNumber.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package beam.sim
```
Comment (JustinPihony) [inline src/main/scala/beam/sim/CustomTerminateAtFixedIterationNumber.scala]: Is this used? Using the Github UI I don't see any actual usage.

Comment (JustinPihony) [reply src/main/scala/beam/sim/CustomTerminateAtFixedIterationNumber.scala]: Or is this used by virtue of extending TerminationCriterion - thus the way binding is done?

File: src/main/java/beam/matsim/MatsimConfigUpdater.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    MatsimConfigUpdater(BeamConfigChangesObservable beamConfigChangesObservable, LinkStatsConfigGroup linkStatsConfigGroup, ControlerConfigGroup controlerConfigGroup) {
+        this.controlerConfigGroup = controlerConfigGroup;
+        this.linkStatsConfigGroup = linkStatsConfigGroup;
+
+        beamConfigChangesObservable.addObserver(this);
+    }
+
+    @Override
+    public void update(Observable observable, Object o) {
+        Tuple2 t = (Tuple2) o;
```
Comment (JustinPihony) [inline src/main/java/beam/matsim/MatsimConfigUpdater.java]: Is this guaranteed to be the objects you are casting? Should cast checks be made?

File: src/main/java/beam/matsim/CustomPlansDumpingImpl.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
+package beam.matsim;
```
Comment (JustinPihony) [inline src/main/java/beam/matsim/CustomPlansDumpingImpl.java]: Same - I don't see this used, but is probably part of matsim injection?

File: src/main/scala/beam/sim/CustomTerminateAtFixedIterationNumber.scala (hunk 2 above)
Comment (irishwithaxe) [reply src/main/scala/beam/sim/CustomTerminateAtFixedIterationNumber.scala]: yes, this is used in BeamHelper to rewrite behaviour implemented in matsim
[here](https://github.com/LBNL-UCB-STI/beam/pull/2151/files?file-filters%5B%5D=.conf&file-filters%5B%5D=.scala#diff-d4f57c2659d44575c5f00b19c7e0449cR163)

File: src/main/java/beam/matsim/CustomPlansDumpingImpl.java (hunk 4 above)
Comment (irishwithaxe) [reply src/main/java/beam/matsim/CustomPlansDumpingImpl.java]: correct, it is used [here](https://github.com/LBNL-UCB-STI/beam/pull/2151/files?file-filters%5B%5D=.conf&file-filters%5B%5D=.scala#diff-d4f57c2659d44575c5f00b19c7e0449cR159)

File: src/main/java/beam/matsim/MatsimConfigUpdater.java (hunk 3 above)
Comment (irishwithaxe) [reply src/main/java/beam/matsim/MatsimConfigUpdater.java]: looks like it is guaranteed, at least the same code appears in other places, but type checks when casting it is not a bad idea.
I will add it

File: src/main/scala/beam/router/BeamSkimmer.scala (hunk 1 above)
Comment (irishwithaxe) [reply src/main/scala/beam/router/BeamSkimmer.scala]: i will fix it

**Comment by irishwithaxe** [comment]
test!

**Comment by irishwithaxe** [comment]
test!

**Comment by irishwithaxe** [comment]
test!

**Comment by irishwithaxe** [comment]
test!

**Comment by irishwithaxe** [comment]
test!

**Comment by JustinPihony** [comment]
@irishwithaxe This should be good to go, but will let you do the merging

**Pull request merged**